packages/govuk-frontend/src/govuk/components/exit-this-page/template.njk

```diff
@@ -15,6 +15,7 @@
     html: params.html if (params.html or params.text) else defaultHtml,
     text: params.text,
     classes: "govuk-button--warning govuk-exit-this-page__button govuk-js-exit-this-page-button",
-    href: params.redirectUrl | default("https://www.bbc.co.uk/weather")
+    href: params.redirectUrl | default("https://www.bbc.co.uk/weather"),
+    attributes: params.button.attributes | default({ 'rel': 'nofollow noreferrer' })
```

Comment:
These values would be overwritten if a user passes any attributes to the button. I wonder if we should set the value to dynamically include `attributes` if it isn't empty.

Reply:
It's on my list, but not yet.

Reply:
I mentioned it in my thoughts section, yeah. 

I'm not really sure how to deal with that, though. To my knowledge there isn't any easy way to manipulate, merge or add new keys to objects within out-the-box Nunjuck templating, it requires either a custom filter or global function.

We have custom filters in the review app, but as far as I'm aware we've avoided using custom Nunjucks filters in components as it'd require services to have the same filters configured in their own Nunjucks set up.

Reply:
> We have custom filters in the review app, but as far as I'm aware we've avoided using custom Nunjucks filters in components as it'd require services to have the same filters configured in their own Nunjucks set up.

Exactly this. We also want to make it possible for someone to recreate the templates with the same 'API' in other templating languages, which may not support object merging (either natively or by some sort of extension). So we try and restrict ourselves to simple logic and looping functionality that's likely to be universally available.

Has anyone requested being able to pass attributes to the button? Wondering if we should do the simplest thing and just hard code the attributes we need, and not provide macro options for them?

```js
attributes: {
  "rel": "nofollow noreferrer"
}
```

Reply:
It's not been a request from a user, no, but the lack of user-definable attributes was noted as a barrier to users patching the issue manually—it wouldn't have required a new code release to fix and we could have issued guidance in the interim.

Making the attribute configurable is a bit of pre-emptive futureproofing in that way, but I suppose there is no requirement we actually do it if we don't think it'll be useful.

Reply:
It was my suggestion off of the linked issue to include attributes. As we're finding out, this might not be feasible. I've been running this over for a day and can't come up with a way to get this to work within our constraints. I'm leaning towards hard coding the attributes as well.

The one hypothetical use case for allowing attributes is click tracking, which would be useful for us in terms of gathering usage within services. I don't think that's a show stopper right now until we solve this thorny issue of how to manipulate existing objects though. On top of that, nobody's asked for it yet.

I can see there's some merge discourse going on in the thread below.

Reply:
I've opted to hardcode the attributes on the basis that it leaves room for us to make them configurable at a later date without introducing a breaking change. If we committed to one merge/override strategy now, and changed it later down the line, that would be much more troublesome.